packages/blade-old/scripts/icon-migrate-codemod/cli.js

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env node
+const path = require('path');
+const execa = require('execa');
+const argParser = require('args-parser');
+const jscodeshiftExecutable = require.resolve('.bin/jscodeshift');
+const transformerDirectory = path.resolve(__dirname, './transforms');
+
+function run() {
+  const transformerPath = path.join(transformerDirectory, `blade-old-icon.ts`);
+
+  const parsedArgs = argParser(process.argv);
+  const input = process.argv.slice(2)[0];
+  const args = [input, '-t', transformerPath, ...process.argv.slice(3)];
+  const parser = parsedArgs.parser || 'jsx';
+
+  if (parser === 'tsx') {
+    args.push('--extensions=tsx,ts,jsx,js');
```

Comment:
why do we need tsx? for us default parse is jsx only right since everything in old-blade is jsx?

Reply:
It's for the applications which will be using this codemod, if they are on typescript they need to pass the parser arg in the cli `parser=jsx` so that jscodeshift can handle it.